### PR TITLE
ao_oss: use --audio-device if --oss-device isn't set.

### LINKED ISSUE #4122: [regression] --audio-device doesn't work with ao=oss
The argument ```-ao oss:device=/dev/dspN``` used to work. At some point options changed (https://github.com/mpv-player/mpv/wiki/Option-replacement-list). It says ```--ao=oss:device => --audio-device (different semantics)```, but ```--ao=oss --audio-device=/dev/dspN``` doesn't play to that device.

### PR DESCRIPTION
**Pull request by kevmitch**
Fall back on PATH_DEV_DSP if nothing is set.

This mirrors the behaviour of --audio-device / --alsa-device.

There doesn't appear to be a general way to list devices with oss, so
--audio-device=help doesn't list oss devices except for the default one if the
file exists.

Previously --audio-device was ignored entirely by ao_oss.

fixes #4122

I agree that my changes can be relicensed to LGPL 2.1 or later.

### CODE REVIEW

**Comment by wm4** [comment]
OK. That `if` could probably be cosmetically collapsed to 2 lines with `?:`, but doesn't matter too much.

**Comment by kevmitch** [comment]
avoid setting the p->dsp option value

**Comment by wm4** [comment]
Feel free to push.

File: audio/out/ao_oss.c [hunk 1]
```diff
@@ -279,14 +279,21 @@ static int reopen_device(struct ao *ao, bool allow_format_changes)
     int format = ao->format;
     struct mp_chmap channels = ao->channels;
 
+    const char *device = PATH_DEV_DSP;
+    if (ao->device)
+        device = ao->device;
```
Comment (wm4) [inline audio/out/ao_oss.c]: Just randomly looked at this again. Sure that you don't need to check `ao->device[0]`? Or maybe that's intentional.

Comment (kevmitch) [inline audio/out/ao_oss.c]: This was a semi-robotic copy from ao_alsa.c. My solution for both is  #4176.

**Comment by kevmitch** [comment]
this was merged

**Comment by wm4** [comment]
It was? Whoops.

**Pull request closed without merging**